sentinelhub/api/batch_statistical.py

```diff
@@ -0,0 +1,148 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+from dataclasses import dataclass, field
+from typing import Any, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ..type_utils import Json, JsonDict
+from .batch_base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+from .utils import datetime_config, enum_config, remove_undefined
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class SentinelHubBatchStatistical(BaseBatchClient["BatchStatisticalRequest"]):
+    """An interface class for Sentinel Hub Batch Statistical API
+
+    Check `Batch Statistical API <https://docs.sentinel-hub.com/api/latest/reference/#tag/batch_statistical>`__ for more
+    information.
+    """
+
+    # pylint: disable=too-many-public-methods
+    @staticmethod
+    def _get_service_url(base_url: str) -> str:
+        """Provides URL to Batch Statistical API"""
+        return f"{base_url}/api/v1/statistics/batch"
+
+    def create(
+        self,
+        input_data: JsonDict,
+        input_features: JsonDict,
+        aggregation: JsonDict,
+        calculations: JsonDict,
```

Comment:
I'm starting to wonder if perhaps it would be better to replace `input_data`, `aggregation`, and `calculations` with `statistical_request`. This would be more in-line with `SentinelHubBatch`. :thinking:

Reply:
this was part of the discussion we had previously. I was thinking that we could perhaps add a builder method that would work with `statistical_request` +  S3 params?

Reply:
I would still leave `input_features` because `s3` might not be the only option ever supported. But maybe `(statistical_request, input_features, output, **kwargs)` could work :thinking:

Reply:
done